pkg/sidecar/sidecarpb/slicegateway_handler.go

```diff
@@ -66,31 +59,22 @@ func sliceGwSetInterClusterDscpConfig(dscpClass string) error {
 		portFilter = "--source-port 11194"
 	}
 
-	if interClusterDscpClass == dscpClsStringToDscpCodeMap[dscpClass] {
+	if interClusterDscpClass == dscpClass {
 		log.Infof("No change in DSCP marking needed: %v", interClusterDscpClass)
 		return nil
 	}
-	// Delete existing DSCP config before adding a new one
-	if interClusterDscpClass != "Default" {
-		ipTablesCmd := fmt.Sprintf("iptables -t mangle -D POSTROUTING -p udp %s -j DSCP --set-dscp-class %s",
-			portFilter, interClusterDscpClass)
-		_, err := runCommand(ipTablesCmd)
+	var err error
+	if !containsDscp(userDscpClasses, dscpClass) {
```

Comment:
There can only be one dscp class setting at any point of time. userDscpClasses as an array is not required. Please change it to a string type and call it dscpClassInUse.

If there is a change in the dscp class, we need to delete the existing entry from the iptables mangle table. So you will have to keep the previous implementation that deleted the entry first before adding a new entry.